### DISCUSSION TITLE
Exclude linelength rule from google-java-format?

### DISCUSSION THREAD

**Discussion by tisonkun** [diffplug/spotless#1645] [Q&A]
I'm satisfied with GJF except linelength ≤ 80 is too strict for me.

I wonder if I can apply GJF with one exception - no limit on linelength or extend to 180?

cc @nedtwigg

**Comment by jbduncan** [comment]
Welcome, @tisonkun!

This is a known and intentional limitation of GJF. See their [FAQ](https://github.com/google/google-java-format/wiki/FAQ.)  for more information. (On reading it, it seems to say that the max line length is actually 100, interestingly enough.)

Sorry I couldn't give you a satisfying answer. :(

**Reply by jbduncan** [reply]
If you need longer line lengths, consider palantir-java-format which has a 120 max line length, or Eclipse JDT, which has all the flexibility of Eclipse IDE's own formatter but needs more setup (although you can follow these [instructions](https://github.com/diffplug/spotless/blob/main/ECLIPSE_SCREENSHOTS.md) if you wish, or use [JUnit 5's config files](https://github.com/junit-team/junit5/tree/main/gradle/config/eclipse) as a base as long as you follow their [open source license](https://github.com/junit-team/junit5/blob/main/LICENSE.md).)

I hope this helps.
